pygtracker/pygtracker.py

```diff
@@ -126,7 +126,58 @@ def rank_students(self, course_id="all", n=10, descending=True):
                 rank: int
                 grade: float
         """
-        return None
+        # check if ascending is a boolean
+        if not isinstance(ascending, bool):
+            raise TypeError("Ascending value should be a boolean.")
+            # check if course id is a string
+        if not isinstance(course_id, str):
+            raise TypeError("Course id should be a string.")
+            # check if n is an integer and is less than the total
+        if not isinstance(n, int):
+            raise TypeError("N value should be a integer.")
+
+        # call helper function and get the dataframe
+        course_and_grade_df = self.calculate_final_grade(
+            self.courses["course_id"].unique().tolist()
+        )
+
+        # check if course_id is part of courses list - need to comment when self.courses in
+        if not (
+            course_id in (course_and_grade_df["course_id"].unique().tolist() + ["all"])
+        ):
+            raise NameError("Course ID is not a part of the courses dataset.")
```

Comment:
What is a NameError? A NameError is raised when you try to use a variable or a function name that is not valid.

I think the more appropriate exception here is ValueError. But you need to change the test accordingly
```suggestion
            raise ValueError("Course ID is not a part of the courses dataset.")
```